baseplate/core.py

```diff
@@ -317,19 +324,59 @@ def __exit__(self, exc_type, value, traceback):
         else:
             self.finish()
 
+    def make_child(self, name, local=False, component_name=None):
+        """Return a child Span whose parent is this Span.
 
-class ServerSpan(Span):
-    """A server span represents a request this server is handling.
+        The child span can either be a local span representing an in-request operation
+        or a span representing an outbound service call.
 
-    The server span is available on the :term:`context object` during requests
-    as the ``trace`` attribute.
+        In a server, a local span represents the time spent within a
+        local component performing an operation or set of operations.
+        The local component is some grouping of business logic,
+        which is then split up into operations which could each be wrapped
+        in local spans.
 
-    """
+        """
+        raise NotImplementedError
 
-    def make_child(self, name):
-        """Return a child span representing an outbound service call."""
+
+class LocalSpan(Span):
+    # pylint: disable=invalid-name
+    def __init__(self,
+                 trace_id,
+                 parent_id,
+                 span_id,
+                 sampled,
+                 flags,
+                 name,
+                 context):
+        super(LocalSpan, self).__init__(trace_id, parent_id, span_id, sampled,
+                                        flags, name, context)
+        self.context = context
+
+    def make_child(self, name, local=False, component_name=None):
         span_id = random.getrandbits(64)
-        span = Span(self.trace_id, self.id, span_id, self.sampled, self.flags, name)
+
+        if local:
+            context_copy = self.context.clone()
+            span = LocalSpan(self.trace_id, self.id, span_id, self.sampled,
+                             self.flags, name, context_copy)
+            if component_name is None:
+                raise SpanError("Cannot create local span without component name.")
+            span.component_name = component_name
+            context_copy.trace = span
```

Comment:
It's getting passed into the span (https://github.com/reddit/baseplate/pull/75/files/b92742b01897b2921a5ae78f065b5ee0dab6a8b9#diff-e23a2f0193db987b0509d66964675524R362) which is how the copied context will be accessed within the span block.